### PR TITLE
Added entity and repository classes for users

### PR DESCRIPTION
**Pull request by AlexanderK88**
### **Description**:
This pull request introduces the essential **entity** and **repository** classes to support the implementation of permissions, roles, and users within the application. These additions lay the groundwork for a role-based access control (RBAC) system.

---

#### **Key Changes**:

1. **Entities**:
   
   - **`User.java`**:
     - Defines the `User` entity with fields:
       - `id`: Primary key.
       - `name`: User's full name.
       - `email`: User's email address.
       - `password`: User's password.

2. **Repositories**:
   - **`UserRepository.java`**:
     - Provides CRUD operations for the `User` entity.

---

#### **Purpose**:
- Establishes the foundational data models required for managing users.
- Enables future development of security features and access control mechanisms.
- Facilitates database interactions through Spring Data repositories.

### CODE REVIEW

File: src/main/java/org/fungover/system2024/repositories/UserRepository.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package org.fungover.system2024.repositories;
+
+import org.fungover.system2024.entities.User;
+import org.springframework.data.repository.ListCrudRepository;
```
[the remark this answers is not in the document]
Comment (MikaelaMathisson) [reply src/main/java/org/fungover/system2024/repositories/UserRepository.java:4]: Look into this?

**Review by MikaelaMathisson — COMMENTED** [review]
I think the code looks good, and does what it should. Clean code thats easy to read through. I see that the code review bot had some things to say, maybe its good to look into if theres anything that needs to be changed for the better.

**Review by sebastianpiresmolin — APPROVED** [review]
Pretty straightforward and fulfills what the issue is aimed at achieving.

As a side note, maybe consider adding some constraints to the entity? (@Size or @Length for example)

Should be able to go ahead and merge this.

File: src/main/java/org/fungover/system2024/entities/Permission.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", nullable = false)
+    private Integer id;
+
+    @Column(name = "name", nullable = false)
+    private String name;
+
+    @Lob
+    @Column(name = "description")
+    private String description;
```
[the remark this answers is not in the document]
Comment (AlexanderK88) [reply src/main/java/org/fungover/system2024/entities/Permission.java]: File removed

File: src/main/java/org/fungover/system2024/entities/Permission.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    @ManyToMany
+    @JoinTable(name = "role_permissions",
+            joinColumns = @JoinColumn(name = "permission_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id"))
+    private Set<org.fungover.system2024.entities.Role> roles = new LinkedHashSet<>();
```
[the remark this answers is not in the document]
Comment (AlexanderK88) [reply src/main/java/org/fungover/system2024/entities/Permission.java]: File removed

File: src/main/java/org/fungover/system2024/entities/Role.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package org.fungover.system2024.entities;
+
+import jakarta.persistence.*;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
```
[the remark this answers is not in the document]
Comment (AlexanderK88) [reply src/main/java/org/fungover/system2024/entities/Role.java]: File removed

File: src/main/java/org/fungover/system2024/entities/RolePermission.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    @EmbeddedId
+    private RolePermissionId id;
+
+    @MapsId("permissionId")
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @JoinColumn(name = "permission_id", nullable = false)
+    private org.fungover.system2024.entities.Permission permission;
+
+    @MapsId("roleId")
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
```
[the remark this answers is not in the document]
Comment (AlexanderK88) [reply src/main/java/org/fungover/system2024/entities/RolePermission.java]: File removed

File: src/main/java/org/fungover/system2024/entities/Role.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import jakarta.persistence.*;
+import lombok.Getter;
+import lombok.Setter;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+
+@Getter
+@Setter
+@Entity
+@Table(name = "roles", schema = "system24db")
```
[the remark this answers is not in the document]
Comment (AlexanderK88) [reply src/main/java/org/fungover/system2024/entities/Role.java]: File removed

File: src/main/java/org/fungover/system2024/entities/Role.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", nullable = false)
+    private Integer id;
+
+    @Size (max = 255)
+    @NotNull
+    @Column(name = "name", nullable = false)
+    private String name;
```
[the remark this answers is not in the document]
Comment (AlexanderK88) [reply src/main/java/org/fungover/system2024/entities/Role.java]: File removed

File: src/main/java/org/fungover/system2024/repositories/UserRepository.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (FelixHultman) [reply src/main/java/org/fungover/system2024/repositories/UserRepository.java:4]: I think the bot is not up to date with spring 3.

**Review by AlbinNils98 — COMMENTED** [review]
I noticed when working on my issue #14 that, as the bot here says, the user entity has both email and password named as column "email". This makes it so the application can't start as it introduces duplication problems for the database. 

Another thing i think would be appropriate is to name the tables in the database in singular e.g. user instead of users as it makes it more comprehensive when working with the tables, think user.name and users.name, the first one feels to me more reasonable.  
Of course there is no true convention to follow but we should choose one or the other.

**Review by AlbinNils98 — APPROVED** [review]
After the recent changes i think the code looks great!

**Review by sebastianpiresmolin — APPROVED** [review]
I see that you have added constrains to you entity, as well as some minor changes to naming convention from feedback you have got in this PR. Great job guys.

**Review by MasakiShiraishi — APPROVED** [review]

**Review by AlbinNils98 — APPROVED** [review]

**Review by sebastianpiresmolin — APPROVED** [review]

**Pull request merged**